primer-miso/build-frontend.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ $# -eq 0 ]; then
+    echo "Building for dev"
+    dev_mode=true
+else
+    echo "Building for prod"
+    dev_mode=false
+fi
+
+rm -rf dist
+mkdir dist
+cp frontend/*.html dist/
+
+hs_wasm_path=$(wasm32-wasi-cabal list-bin -v0 exe:primer-miso)
+
+ghc_wasm_jsffi="dist/ghc_wasm_jsffi.js"
+
+"$(wasm32-wasi-ghc --print-libdir)"/post-link.mjs \
+                                   --input "$hs_wasm_path" --output "$ghc_wasm_jsffi"
+
+if ! [ -f "$ghc_wasm_jsffi" ] ; then
+    echo "post-link.mjs didn't produce a $ghc_wasm_jsffi file. Make sure you're in the Nix Wasm shell."
+    exit 1
+fi
+
+if $dev_mode; then
+    cp "$hs_wasm_path" dist/bin.wasm
+else
+    wizer --allow-wasi --wasm-bulk-memory true --init-func _initialize -o dist/bin.wasm "$hs_wasm_path"
+    wasm-opt ${1+"$@"} dist/bin.wasm -o dist/bin.wasm
+    wasm-tools strip -o dist/bin.wasm dist/bin.wasm
+    brotli --rm --best dist/bin.wasm -o dist/bin.wasm.br
+    mv dist/bin.wasm.br dist/bin.wasm
```

Comment:
This doesn't work. Try changing `serve-frontend: frontend` to `serve-frontend: frontend-prod` in the makefile and you'll see the frontend fail to load, with something like `Uncaught CompileError: wasm validation error: at offset 4: failed to match magic number` in the browser console.

I think you need to set the right content headers somewhere. The ones which most servers enable by default for `gzip`.

Reply:
AFAICT, `simple-http-server` doesn't support this. But `http-server` from `npm` does: `nix shell nixpkgs/24.05#http-server -c http-server -b -p 8000 dist`.

Reply:
Also, I think one has to keep the `.br` suffix, or at least it's best practice (leaving it off does work in some cases for `gzip` at least).

Reply:
(FYI, none of these problems are present on macOS, where the `prod` build works fine and serves up `.br` files without the suffix.)

Reply:
Oh, weird. Maybe I should double check the behaviour on my end. I wouldn't have expected this to be an OS difference.

Reply:
I'm now failing to reproduce this, so let's assume it's not a problem (although I don't currently have access to the same machine on which I originally had the issue).